lib/engine/game/g_1847_ae/game.rb

```diff
@@ -111,6 +123,57 @@ def operating_round(round_num)
             [Engine::Step::BuyCompany, { blocks: true }],
           ], round_num: round_num)
         end
+
+        def saar
+          corporation_by_id('Saar')
+        end
+
+        def hlb
+          corporation_by_id('HLB')
+        end
+
+        def init_share_pool
+          G1847AE::SharePool.new(self)
+        end
+
+        # Reserved share type used to represent investor shares
+        def ipo_reserved_name(_entity = nil)
+          'Investor'
+        end
+
+        def init_corporations(stock_market)
+          self.class::CORPORATIONS.map do |corporation|
+            par_price = stock_market.par_prices.find { |p| p.price == corporation[:required_par_price] }
+            corporation[:par_price] = par_price
+            G1847AE::Corporation.new(
+              min_price: par_price.price,
+              capitalization: self.class::CAPITALIZATION,
+              **corporation.merge(corporation_opts),
+            )
+          end
```

Comment:
no need to do this, define CORPORATION_CLASS instead

Reply:
I have to pass the par_price object to the Corporation. And I can't reach @game (it's nil) from inside the Corporation.initialize from what I see.

Reply:
you can always set min_price in setup